ThirdPartyModules/Amasty/Rewards.php

```diff
@@ -103,4 +106,59 @@ public function getAdditionalJS($result)
         }';
         return $result;
     }
+    
+    /**
+     * Return code if the quote has Amasty reward points.
+     * 
+     * @param $result
+     * @param $couponCode
+     * @param $quote
+     * 
+     * @return array
+     */
+    public function filterVerifyAppliedStoreCredit (
+        $result,
+        $couponCode,
+        $quote
+    )
+    {
+        if ($couponCode == self::AMASTY_REWARD && $quote->getData('amrewards_point')) {
+            $result[] = $couponCode;
+        }
+        
+        return $result;
+    }
+    
+    /**
+     * Remove Amasty reward points from the quote.
+     *
+     * @param $amastyRewardsManagement
+     * @param $couponCode
+     * @param $quote
+     * @param $websiteId
+     * @param $storeId
+     * 
+     */
+    public function removeAppliedStoreCredit (
+        $amastyRewardsManagement,
+        $amastyRewardsQuote,
+        $couponCode,
+        $quote,
+        $websiteId,
+        $storeId
+    )
+    {
+        try {
+            if ($couponCode == self::AMASTY_REWARD && $quote->getData('amrewards_point')) {
+                $amastyRewardsManagement->collectCurrentTotals($quote, 0);
+
+                $amastyRewardsQuote->addReward(
+                    $quote->getId(),
+                    0
+                );
+            }
+        } catch (\Exception $e) {
+            throw $e;
```

Comment:
Can you log the exception to Bugsnag?

Reply:
@johnmccombs1 it would sent to bugsnag via https://github.com/BoltApp/bolt-magento2/blob/master/Model/Api/UpdateDiscountTrait.php#L474